lib/install.mjs

```diff
@@ -0,0 +1,193 @@
+import { createWriteStream, existsSync } from 'node:fs';
+import { rename, rm } from 'node:fs/promises';
+import { get } from 'node:https';
+import { dirname, resolve } from 'node:path';
+import { arch, env, platform, exit } from 'node:process';
+import { fileURLToPath, URL } from 'node:url';
+
+import compressing from 'compressing';
+import progress from 'cli-progress';
+import semver from 'semver';
+
+import nodeManifest from '../package.json' assert { type: 'json' };
+
+/**
+ * NW.js build flavor
+ * 
+ * @type {'sdk' | 'normal'}
+ */
+let buildType = env.npm_config_nwjs_build_type || env.NWJS_BUILD_TYPE || 'normal';
+
+/**
+ * Parsed string version to Semver version object
+ */
+let parsedVersion = semver.parse(nodeManifest.version);
+
+/**
+ * Version string of the format `X.Y.Z-pre`.
+ * The prerelease segment `pre` is used to specify build flavor, or other tags.
+ * 
+ * @type {string}
+ */
+let versionString = [
+  parsedVersion.major,
+  parsedVersion.minor,
+  parsedVersion.patch
+].join('.');
+
+// Check if version is a prelease.
+if (typeof parsedVersion?.prerelease?.[0] === 'string') {
+  let prerelease = parsedVersion.prerelease[0].split('-');
+  if (prerelease.length > 1) {
+    prerelease = prerelease.slice(0, -1);
+  }
+  versionString = [versionString, prerelease].join('-');
```

Comment:
Prelease is converted to an array, previously it was being joined with a `-` before being combined. This should correct the code to match the original.
```suggestion
  versionString = [versionString, ...prerelease].join('-');
```

Reply:
Ahh because prerelease is an `Array<number | string>`!